compiler/semantic/op.go

```diff
@@ -78,42 +74,79 @@ func wrapAlias(alias string, seq dag.Seq) dag.Seq {
 	})
 }
 
-func (a *analyzer) semFromEntity(entity ast.FromEntity, args ast.FromArgs) dag.Seq {
+func (a *analyzer) semFromEntity(entity ast.FromEntity, args ast.FromArgs, seq dag.Seq) dag.Seq {
 	switch entity := entity.(type) {
 	case *ast.Glob:
+		if a.hasFromParent(entity, seq) {
+			return seq
+		}
```

Comment:
Nit: Returning seq when hasFromParent failes doesn't really make sense. Clearer to return dag.Seq{badOp()}.

Reply:
Well that pattern that is used throughout it to march along with the bad seq's.. I thought this was useful for debugging but I will change this.

Reply:
I guess what I'm getting at is that seq is here for use in analysis but isn't meant to be returned. Even the successful returns from this method don't incorporate seq.